src/core/server/legacy/legacy_service.ts

```diff
@@ -188,51 +175,4 @@ export class LegacyService implements CoreService {
 
     return kbnServer;
   }
-
-  private setupProxyListener(server: HapiServer) {
-    const legacyProxy = new LegacyPlatformProxy(
-      this.coreContext.logger.get('legacy-proxy'),
-      server.listener
-    );
-
-    // We register Kibana proxy middleware right before we start server to allow
-    // all new platform plugins register their routes, so that `legacyProxy`
-    // handles only requests that aren't handled by the new platform.
-    server.route({
-      path: '/{p*}',
-      method: '*',
-      options: {
-        payload: {
-          output: 'stream',
-          parse: false,
-          timeout: false,
-          // Having such a large value here will allow legacy routes to override
-          // maximum allowed payload size set in the core http server if needed.
-          maxBytes: Number.MAX_SAFE_INTEGER,
-        },
-      },
-      handler: async ({ raw: { req, res } }, responseToolkit) => {
-        if (this.kbnServer === undefined) {
-          this.log.debug(`Kibana server is not ready yet ${req.method}:${req.url}.`);
-
-          // If legacy server is not ready yet (e.g. it's still in optimization phase),
-          // we should let client know that and ask to retry after 30 seconds.
-          return responseToolkit
-            .response('Kibana server is not ready yet')
-            .code(503)
-            .header('Retry-After', '30');
```

Comment:
should we put this logic back?

Reply:
```js
if (setupDeps.core.http.shouldListen()) {
  const tempServer = new Server(setupDeps.core.http.options);
  tempServer.route({
    path: '/{p*}',
    method: '*',
    handler: (req, responseToolkit) => {
      this.log.debug(`Kibana server is not ready yet ${req.method}:${req.url}.`);

      // If legacy server is not ready yet (e.g. it's still in optimization phase),
      // we should let client know that and ask to retry after 30 seconds.
      return responseToolkit
        .response('Kibana server is not ready yet')
        .code(503)
        .header('Retry-After', '30');
    },
  });
  await tempServer.start();

  const kbnServer = await this.createKbnServer(config, setupDeps, startDeps);
  await tempServer.stop();
  return kbnServer;
} else {
  return await this.createKbnServer(config, setupDeps, startDeps);
}
```

Reply:
I think we should put this behavior back somewhere. We used to start up Kibana pretty much immediately, but now we have blocking behaviors during startup for things like saved object migration which makes it easy to hit this server-not-ready error in practice. I believe this behavior was added after we rolled out the legacy service.

Reply:
I agree, we should keep this behavior in one form or another, I'm fine with handling this in an immediate follow-up though.

Reply:
put back  as a part of HttpService